File: server/router/kubernetes/daemonSet.go
package kubernetes

import (
	"github.com/gin-gonic/gin"
	v1 "github.com/pddzl/kubeops/server/api/v1"
	"github.com/pddzl/kubeops/server/middleware"
)

type DaemonSetRouter struct{}

func (d *DaemonSetRouter) InitDaemonSetRouter(Router *gin.RouterGroup) {
	daemonSetRouter := Router.Group("daemonSet").Use(middleware.OperationRecord())
	daemonSetRouterWithoutRecord := Router.Group("daemonSet")
	daemonSetApi := v1.ApiGroupApp.KubernetesApiGroup.DaemonSetApi
	{
		daemonSetRouter.POST("getDaemonSetDetail", daemonSetApi.GetDaemonSetDetail) // 获取daemonSet详情
		daemonSetRouter.POST("getDaemonSetRaw", daemonSetApi.GetDaemonSetRaw)       // 获取daemonSet in 编排
		daemonSetRouter.POST("getDaemonSetPods", daemonSetApi.GetDaemonSetPods)     // 获取daemonSet关联的pods
	}
	{
		daemonSetRouterWithoutRecord.POST("getDaemonSetList", daemonSetApi.GetDaemonSetList) // 获取所有daemonSet
	}
}

File: web/src/api/kubernetes/daemonSet.js
import service from '@/utils/request'

// 获取daemonSet list
export const getDaemonSetList = (data) => {
  return service({
    url: '/daemonSet/getDaemonSetList',
    method: 'post',
    data
  })
}

// 获取daemonSet in raw
export const getDaemonSetRaw = (data) => {
  return service({
    url: '/daemonSet/getDaemonSetRaw',
    method: 'post',
    data
  })
}

// 获取daemonSet in detail
export const getDaemonSetDetail = (data) => {
  return service({
    url: '/daemonSet/getDaemonSetDetail',
    method: 'post',
    data
  })
}

// 获取daemonSet关联pods
export const getDaemonSetPods = (data) => {
  return service({
    url: '/daemonSet/getDaemonSetPods',
    method: 'post',
    data
  })
}

// 删除daemonSet
export const deleteDaemonSet = (data) => {
  return service({
    url: '/daemonSet/deleteDaemonSet',
    method: 'post',
    data
  })
}
